store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreConfiguredGraphIdsTest.java

```diff
@@ -0,0 +1,299 @@
+                .op(new GetSchema.Builder()
+                        .compact(true)
+                        .build())
+                .graphIds(Collections.singletonList(GRAPH_ID_SELECTED))
+                .build();
+        final Schema result = (Schema) federatedStore.execute(federatedOperation, contextTestUser());
+
+        // Then
+        assertNotNull(result);
+        JsonAssert.assertEquals(selectedSchema.toJson(true), result.toJson(true));
+    }
+
+    @Test
+    void shouldReturnUnselectedGraphIdSchemaIfUnselectedGraphIdSet()
+            throws OperationException {
+        // Given
+        final FederatedStore federatedStore = getFederatedStoreFromProperties(FEDERATED_STORE_CONFIGURED_GRAPH_IDS_PROPERTIES);
+        addSelectedSchemaAndGraph(federatedStore);
+        final Schema unselectedSchema = addUnselectedSchemaAndGraph(federatedStore);
+
+        // When
+        final FederatedOperation<Object, Object> federatedOperation = new FederatedOperation.Builder()
+                .op(new GetSchema.Builder()
+                        .compact(true)
+                        .build())
+                .graphIds(Collections.singletonList(GRAPH_ID_UNSELECTED))
+                .build();
+        final Schema result = (Schema) federatedStore.execute(federatedOperation, contextTestUser());
+
+        // Then
+        assertNotNull(result);
+        JsonAssert.assertEquals(unselectedSchema.toJson(true), result.toJson(true));
+    }
+
+    // TODO: look into and remove test or comment before merging
+    @Disabled("Fails. Could be a bug with GetSchemaHandler")
+    @Test
+    void shouldReturnSelectedGraphIdSchemaOnlyIfGraphIdsNotSet()
+            throws OperationException {
+        // Given
+        final FederatedStore federatedStore = getFederatedStoreFromProperties(FEDERATED_STORE_CONFIGURED_GRAPH_IDS_PROPERTIES);
+        final Schema selectedSchema = addSelectedSchemaAndGraph(federatedStore);
+        addUnselectedSchemaAndGraph(federatedStore);
+
+        // When
+        final FederatedOperation<Object, Object> federatedOperation = new FederatedOperation.Builder()
+                .op(new GetSchema.Builder()
+                        .compact(true)
+                        .build())
+                .build();
+        final Schema result = (Schema) federatedStore.execute(federatedOperation, contextTestUser());
+
+        // Then
+        assertNotNull(result); // TODO: Fails as result == null
+        JsonAssert.assertEquals(selectedSchema.toJson(true), result.toJson(true));
+    }
+
+    private static FederatedStore getFederatedStoreFromJson(final String path) throws IOException, StoreException {
+        final FederatedStore federatedStore = loadFederatedStoreFrom(path);
+        federatedStore.initialise(GRAPH_ID_SELECTED, new Schema(), new FederatedStoreProperties());
+        addDefaultLibrary(federatedStore);
+        return federatedStore;
+    }
```

Comment:
This seems like quite a generic utility test method for federated stores. In fact, looking at the `FederatedStoreDefaultGraphsTest.java` file which was deleted in this PR, there's this in there:

```
import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.loadFederatedStoreFrom;
...
FederatedStore defaultedFederatedStore = loadFederatedStoreFrom("configuredGraphIds.json");
```

This new method appears to be a duplicate of that method.

Reply:
`FederatedStoreDefaultGraphsTest.java` was not deleted but renamed but it seems that git didn't pick up on this. Just wondering whether any other test class would use the test methods if they were added to the test util class. That is why I kept them in the `FederatedStoreConfiguredGraphIdsTest`

Reply:
I added a method to create the Federated Store as there is now the loading from a properties as well as the loading from `JSON` method mentioned.

Reply:
Ah ok, shame it doesn't recognise that - I've seen this before when there are too many changes. I think doing `git mv` and then making a commit straight away with just the rename _might_ help this in future.
If the methods were already in this class before the rename then don't worry about changing that. I'd assumed they were new. If the are new then it would make more sense to put them in the utility class, but you're right that nothing else would be using them.

It might be better to have a new ticket for refactoring all federation tests to ensure there are no duplicating test methods and that everything is in the test utility class.